components/description-list/description-list-wrapper.js

```diff
@@ -0,0 +1,100 @@
+import { _generateBodyCompactStyles, _generateLabelStyles } from '../typography/styles.js';
+import { css, html, LitElement } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const descriptionListStyles = [
+	_generateLabelStyles('dt'),
+	_generateBodyCompactStyles('dd'),
+	css`
+		:host {
+			--d2l-dt-min-width: min-content;
+			--d2l-dt-max-width: 10rem;
+			--d2l-dd-min-width: 50%;
+		}
+		dl {
+			align-items: baseline;
+			display: var(--d2l-description-list-dl-display, grid);
+			gap: 0.5rem 1.5rem;
+			grid-auto-flow: row;
+			grid-template-columns: minmax(var(--d2l-dt-min-width), auto) minmax(var(--d2l-dd-min-width), 1fr);
+		}
+		dt {
+			max-width: var(--d2l-dt-max-width);
+		}
+		dd {
+			margin: var(--d2l-description-list-dd-margin, 0);
+		}
+	`,
+];
+
+/**
+ * Wraps a native <dl> element, providing styling and resize behavior.
+ * @slot - Content to wrap
+ */
+class DescriptionListWrapper extends LitElement {
+	static get properties() {
+		return {
+			breakpoint: { type: Number, reflect: true },
+			_stacked: { state: true },
+
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.stacked {
+				--d2l-description-list-dl-display: block;
+				--d2l-dt-max-width: unset;
+				--d2l-description-list-dd-margin: 0 0 0.5rem 0;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this._stacked = false;
+		this.breakpoint = 350;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	render() {
+		const classes = {
+			'stacked': this._stacked,
+		};
+		return html`<slot class="${classMap(classes)}"></slot>`;
+	}
+
+	_onResize(entries) {
+		if (!entries || entries.length === 0) return;
+		const entry = entries[0];
+		this._contentWidth = entry.contentRect.width;
+		this._updateStacked();
```

Comment:
Since `_updateStacked()` is now only called from here, might be reasonable to just move it into here which would eliminate the need for the `_contentWidth` member variable.